ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/Ledger.hs

```diff
@@ -117,53 +120,110 @@ instance CanHardFork xs => IsLedger (LedgerState (HardForkBlock xs)) where
 
   type AuxLedgerEvent (LedgerState (HardForkBlock xs)) = OneEraLedgerEvent xs
 
-  applyChainTickLedgerResult cfg@HardForkLedgerConfig{..} slot (HardForkLedgerState st) =
-      sequenceHardForkState
-        (hcizipWith proxySingle (tickOne ei slot) cfgs extended) <&> \l' ->
+  applyChainTickLedgerResult cfg slot lst@(HardForkLedgerState st) =
+      ticked <&> \st' ->
       TickedHardForkLedgerState {
           tickedHardForkLedgerStateTransition =
-            -- We are bundling a 'TransitionInfo' with a /ticked/ ledger state,
-            -- but /derive/ that 'TransitionInfo' from the /unticked/  (albeit
-            -- extended) state. That requires justification. Three cases:
-            --
-            -- o 'TransitionUnknown'. If the transition is unknown, then it
-            --   cannot become known due to ticking. In this case, we record
-            --   the tip of the ledger, which ticking also does not modify
-            --   (this is an explicit postcondition of 'applyChainTick').
-            -- o 'TransitionKnown'. If the transition to the next epoch is
-            --   already known, then ticking does not change that information.
-            --   It can't be the case that the 'SlotNo' we're ticking to is
-            --   /in/ that next era, because if was, then 'extendToSlot' would
-            --   have extended the telescope further.
-            --   (This does mean however that it is important to use the
-            --   /extended/ ledger state, not the original, to determine the
-            --   'TransitionInfo'.)
-            -- o 'TransitionImpossible'. This has two subcases: either we are
-            --   in the final era, in which case ticking certainly won't be able
-            --   to change that, or we're forecasting, which is simply not
-            --   applicable here.
-            State.mostRecentTransitionInfo cfg extended
-        , tickedHardForkLedgerStatePerEra = l'
+            -- TODO justify
+            case State.match (State.tip st) st' of
+              -- TODO Previously, this might have been 'TransitionKnown' if the
+              -- next era uses 'TriggerHardForkAtEpoch'. It /seems/
+              -- inconsequential that we know a bit less here.
+              Left _  -> TransitionUnknown (ledgerTipSlot lst)
+              Right _ -> State.mostRecentTransitionInfo cfg st
```

Comment:
FTR this is the tricky party. There will be a difference in behavior here e.g. when someone configures `TriggerHardForkAtEpoch` with a non-zero argument. (Ideally we want to remove that functionality, see #416.)

An alternative we discussed was to let the cross-era ticking function return a `TransitionInfo`, but this also seems very ad-hoc.

The other alternative is to use `WhetherTickedOrNot` as in #358.

Reply:
There is movement on #416 due to @nfrisby's work, so it probably makes sense to wait for that.